Typo3 12 compatibility

hojalatheef opened the pull request.

hojalatheef committed:
Started compatibility fixing

hojalatheef committed:
[TASK] Replaced TYPO3_MODE with TYPO3

hojalatheef committed:
[TASK] Replaced ERROR and notices namespaces

hojalatheef committed:
[TASK] Replaced ERROR and notices namespaces

hojalatheef committed:
[BUGFIX] Fixed issues with MessageHelper class

hojalatheef committed:
[TASK] READNE.md documentation corrections

hojalatheef committed:
[TASK] Author details and minimum compatibility details updated

hojalatheef committed:
[TASK] Moved Icons registrations to Configurations folder

hojalatheef committed:
[TASK] Moved Icons registrations to Configurations folder

hojalatheef committed:
[TASK] Moved Icons registrations to Configurations folder

sfroemkenjw committed:
Add individual walls io proxy logger

sfroemkenjw committed:
Use strict types where possible

sfroemkenjw committed:
Replace MessageHelper with TYPO3 logging

sfroemkenjw committed:
Update TYPO3 dependencies

hojalatheef committed:
[TASK] Fixes for WallsService to fix issues

hojalatheef committed:
[TASK] Fixes for WallsService to fix issues

hojalatheef committed:
[TASK] Fixes for WallsService to fix issues

hojalatheef committed:
[TASK] Fixes for WallsService to fix issues

hojalatheef committed:
[TASK] Added Documentation with a known issue regarding the emojie handling

hojalatheef committed:
[TASK] Plugin Configuration class corrections

hojalatheef committed:
[TASK] AddWallsProcessor class corrections

hojalatheef committed:
[TASK] AddWallsProcessor class corrections

hojalatheef committed:
[TASK] Made the property wallsService strictly typed

sfroemkenjw committed:
Implement new testing environment

hojalatheef committed:
[TASK] Code formatting for better readability and changes with property strict typing

hojalatheef committed:
[TASK] Code formatting for better readability and changes with property strict typing

hojalatheef committed:
[TASK] Code formatting for better readability and changes with property strict typing

sfroemkenjw committed:
Add pages cache expire time to page layout preview

sfroemkenjw committed:
Add pages cache expire time to page layout preview

hojalatheef committed:
[TASK] Migrating Testing framework from Nimut to TYPO3 TF

hojalatheef committed:
[TASK] Migrating Testing framework from Nimut to TYPO3 TF

hojalatheef committed:
[TASK] UnitTest Case fully compatible with TYPO3 Testing Framework

hojalatheef committed:
[TASK] UnitTest Case fully compatible with TYPO3 Testing Framework

hojalatheef committed:
[BUGFIX] Replaced ObjectProphecy with MockObject

hojalatheef committed:
[TASK] Functional Test Case fixes

hojalatheef committed:
[TASK] Migrated all broken Nimut Functional Test Cases to TYPO3 Testing Framework

hojalatheef committed:
[TASK] Deleted unwanted dataset files from repo

hojalatheef committed:
[BUGFIX] Corrected LICENCE Badge URL

hojalatheef committed:
[BUGFIX] Corrected LICENCE Badge URL Issue with Repo Link

hojalatheef committed:
[BUGFIX] Corrected LICENCE Badge URL Issue with Repo Link

sfroemkenjw commented on .github/workflows/ci.yml:
```diff
@@ -1,83 +1,51 @@
-# Adapted from https://github.com/TYPO3GmbH/blog/blob/master/.github/workflows/ci.yml
-name: CI
+name: Test walls_io_proxy for TYPO3 12 LTS
```
```suggestion
name: Tests
```

sfroemkenjw commented on Classes/Service/WallsService.php:
```diff
@@ -270,14 +256,15 @@ protected function getSanitizedPost(array $post, PluginConfiguration $pluginConf
-        if (!file_exists($filePath)) {
+        if (!file_exists($filePath) && GeneralUtility::getUrl($resource)) {
             GeneralUtility::writeFile($filePath, GeneralUtility::getUrl($resource));
         }
```
Your code requested external API twice:

```suggestion
        if (
            !file_exists($filePath)
            && $responseContent = GeneralUtility::getUrl($resource)
        ) {
            GeneralUtility::writeFile($filePath, $responseContent);
        }
```

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
```diff
@@ -14,3 +14,119 @@ I'm missing new wall posts
+
+#### Steps to Resolve
+
+1. **Update MySQL Database Charset and Collation**
```
4 spaces indent

sfroemkenjw replied:
It's still 3

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
```diff
@@ -14,3 +14,119 @@ I'm missing new wall posts
+
+### Affected Versions
+
+- All versions up to the current release
```
```suggestion
All versions up to the current release
```

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
```diff
@@ -14,3 +14,119 @@ I'm missing new wall posts
+
+   Update the database connection settings in TYPO3 to use `utf8mb4`:
+
+   - **For TYPO3 v9 and earlier:**
```
4 spaces indent

sfroemkenjw commented on Tests/Functional/Service/WallsServiceTest.php:
```diff
@@ -88,27 +80,46 @@ protected function tearDown(): void
-            'Missing entries to show' => [['data' => ['uid' => 1], 'conf' => ['accessToken' => 'ABC123', 'entriesToLoad' => 24]]],
-            'Missing request type' => [['data' => ['uid' => 1], 'conf' => ['accessToken' => 'ABC123', 'entriesToLoad' => 24, 'entriesToShow' => 8]]],
-            'Invalid request type' => [['data' => ['uid' => 1], 'conf' => ['accessToken' => 'ABC123', 'entriesToLoad' => 24, 'entriesToShow' => 8, 'requestType' => 'foo']]],
+            'Missing entries to show' => [
```
Much better. THX

sfroemkenjw commented on Tests/Functional/ViewHelpers/Be/Security/IsAdministratorViewHelperTest.php:
```diff
@@ -11,39 +11,32 @@
-    protected $testExtensionsToLoad = [
-        'typo3conf/ext/walls_io_proxy',
+    protected array $testExtensionsToLoad = [
+        'jweiland/walls_io_proxy',
```
This is not the packagist name

```suggestion
        'jweiland/walls-io-proxy',
```

hojalatheef committed:
[TASK] Updated Testing Suite and removed docker-compose based testing suite

hojalatheef committed:
[TASK] Updated Testing Suite and removed docker-compose based testing suite

hojalatheef committed:
[TASK] Fixed all issues related to CGL and phpLint

hojalatheef committed:
[TASK] Fixed all issues related to CGL and phpLint

hojalatheef committed:
[TASK] Fixed phpStan Level 6 Fixes

hojalatheef committed:
[TASK] Fixed phpStan Level 6 Fixes

hojalatheef committed:
[BUGFIX] CR Changes

hojalatheef committed:
[TASK] Conflict fix for WallsIoClient Test and MessageHelperTest

hojalatheef committed:
[TASK] Fixes from SF

hojalatheef committed:
[TASK] Fixes from SF

hojalatheef committed:
[TASK] Conflict fixed by adapting new Test Suite environment

hojalatheef committed:
[TASK] Updated TestCases and fixed PHPStan issue

hojalatheef committed:
[BUGFIX] Small fix in UnitTest after conflict type

hojalatheef committed:
[TASK] Removed MessageHelper tests from UnitTest

hojalatheef committed:
[TASK] Final Fixes after Rebase and Conflict Fixes

hojalatheef committed:
[BUGFIX] Added composer normalize as dependency

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
```diff
@@ -14,3 +14,119 @@ I'm missing new wall posts
+                    'password' => 'your_database_password',
+                    'host' => 'your_database_host',
+                    'port' => 3306,
+                    'charset' => 'utf8mb4',
```
charset and collation are not allowed on that level. I'm pretty sure they have to be part of tableoptions

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
````diff
@@ -14,3 +14,119 @@ I'm missing new wall posts
+        Edit your `AdditionalConfiguration.php` file:
+
+        ```php
+        $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections']['Default']['charset'] = 'utf8mb4';
````
missing tableoptions

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
````diff
@@ -14,3 +14,119 @@ I'm missing new wall posts
+        Edit your `additional.php` file:
+
+        ```php
+        $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections']['Default']['charset'] = 'utf8mb4';
````
missing tableoptions

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
```diff
@@ -14,3 +14,119 @@ I'm missing new wall posts
+            'DB' => [
+                'Connections' => [
+                    'Default' => [
+                        'charset' => 'utf8mb4',
```
missing tableoptions

sfroemkenjw commented on README.md:
```diff
@@ -1,18 +1,66 @@
+The **walls_io_proxy** extension for TYPO3 addresses this limitation by eliminating the need for client-side cookies from walls.io. It achieves this by moving all XHR/Ajax requests from walls.io's JavaScript file to the server-side using PHP. This approach ensures that no walls.io cookies are set on the client-side, enhancing privacy and compliance with cookie regulations.
 
-`composer req jweiland/walls-io-proxy`
+Please note, **walls_io_proxy** is not an API and does not use the API of walls.io.
```
Does not use the API of walls.io?!?!?!

https://github.com/jweiland-net/walls_io_proxy/blob/main/Classes/Request/PostsRequest.php#L20

sfroemkenjw commented on README.md:
```diff
@@ -1,18 +1,66 @@
+## Features
+
+- **Server-Side Data Retrieval**: Moves all data retrieval processes to the server-side, avoiding client-side cookies.
+- **Customizable Templates**: Allows you to create your own FluidTemplate for displaying the social wall, providing full control over the design and layout.
```
This is in general correct "full control over the design and layout."

BUT: Each social media provider has its own design restrctions. So, you have to adapt these design rules into your templates.

hojalatheef replied:
@sfroemkenjw : I added the same as a note.

sfroemkenjw commented on README.md:
```diff
@@ -1,18 +1,66 @@
+
+## Limitations
+
+- **No API Integration**: This extension does not integrate with the walls.io API.
```
Are you sure?

sfroemkenjw commented on Tests/Functional/DataProcessing/AddWallsProcessorTest.php:
```diff
@@ -14,60 +14,45 @@
 
     /**
-     * @var AddWallsProcessor
+     * @var WallsService|MockObject|(WallsService&object&MockObject)|(WallsService&MockObject)|(object&MockObject)
```
```suggestion
     * @var WallsService|MockObject
```

sfroemkenjw commented on Tests/Unit/Hook/DataHandlerTest.php:
```diff
@@ -11,44 +11,54 @@
 
     /**
-     * @var DataHandler
+     * @var Registry|MockObject|(object&MockObject)|(Registry&object&MockObject)|(Registry&MockObject)
```
```suggestion
     * @var Registry|MockObject
```

sfroemkenjw commented on Tests/Unit/Hook/DataHandlerTest.php:
```diff
@@ -11,44 +11,54 @@
 
     /**
-     * @var WallsService|ObjectProphecy
+     * @var WallsService|MockObject|(WallsService&MockObject)
```
```suggestion
     * @var WallsService|MockObject
```

sfroemkenjw commented on Tests/Unit/Hook/DataHandlerTest.php:
```diff
@@ -11,44 +11,54 @@
+    protected $wallsServiceMock;
+
+    /**
+     * @var ServerRequest|MockObject|(ServerRequest&MockObject)
```
```suggestion
     * @var ServerRequest|MockObject
```

sfroemkenjw commented on composer.json:
```diff
@@ -47,17 +54,20 @@
 	"config": {
-		"vendor-dir": ".build/vendor",
 		"allow-plugins": {
+			"ergebnis/composer-normalize": true,
```
What is that?

sfroemkenjw commented on ext_localconf.php:
```diff
@@ -1,42 +1,58 @@
-    \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addPageTSConfig(
-        '<INCLUDE_TYPOSCRIPT: source="FILE:EXT:walls_io_proxy/Configuration/TSconfig/ContentElementWizard.tsconfig">'
+    ExtensionManagementUtility::addPageTSConfig(
+        '<INCLUDE_TYPOSCRIPT:source = "FILE:EXT:walls_io_proxy/Configuration/TSconfig/ContentElementWizard.tsconfig" > '
```
Isn't there a new `@include` syntax

sfroemkenjw reviewed (changes requested):
Suddenly clicked approve :-) Sorry. Please update marked lines of code.

hojalatheef committed:
[TASK] Small fixes for 11 backward compatibility

hojalatheef committed:
[BUGFIX] Fixed Stan issue

hojalatheef committed:
[BUGFIX] Documentation mistake correction

hojalatheef committed:
[TASK] Added suggested code from CR

hojalatheef committed:
[TASK] Make PageTs configuration inclusion with new approach

Added default page.tsconfig for TYPO3 12 and include this file in ext_localconf with
version compare to avoid multiple inclusion.

hojalatheef committed:
[TASK] StringUtility beginsWith added

Fix for StringUtility beginsWith fixes php version and TYPO3 version challenges

hojalatheef committed:
Update Tests/Unit/Hook/DataHandlerTest.php

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
[TASK] Removed plugin normalize

hojalatheef committed:
Update Tests/Unit/Hook/DataHandlerTest.php

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Update Tests/Unit/Hook/DataHandlerTest.php

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
[TASK] Added note regarding the fluid template customization

hojalatheef committed:
[TASK] Removed Limitations section

hojalatheef committed:
[BUGFIX] Fixed tableoptions in documentation

hojalatheef committed:
[BUGFIX] Added composer plugin normalize as it is included in CI task

hojalatheef committed:
[BUGFIX] CGL Fixes

hojalatheef committed:
[BUGFIX] Fix phpstan errors

sfroemkenjw commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,49 @@
+        php:
+          - '7.4'
+          - '8.1'
+          - '8.2'
```
```suggestion
          - '7.4'
          - '8.0'
          - '8.1'
          - '8.2'
```

sfroemkenjw commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,49 @@
+        run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -d mysql -s functional
+
+      - name: 'Execute functional tests'
+        run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -d mariadb -s functional
```
MariaDB and MySQL are nearly the same. In that extension are no "specials" like in maps2. Let us better try testing on SQLite and additionally on Postgres

```suggestion
      - name: 'Execute functional tests on SQLite'
        run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -d sqlite -s functional

      - name: 'Execute functional tests on MariaDB'
        run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -d mariadb -s functional

      - name: 'Execute functional tests on Postgres'
        run: Build/Scripts/runTests.sh -p ${{ matrix.php }} -d postgres -s functional

```

sfroemkenjw commented on Classes/Request/AbstractRequest.php:
```diff
@@ -18,12 +18,20 @@
 abstract class AbstractRequest implements RequestInterface
 {
+    /**
+     * @var array<string, mixed> $allowedParameters
```
```suggestion
     * @var array<string, mixed>
```

sfroemkenjw commented on Classes/Service/WallsService.php:
```diff
@@ -55,12 +56,18 @@ class WallsService
     protected WallsIoClient $client;
 
-    public function __construct(Registry $registry, WallsIoClient $client)
+    protected ServerRequest $request;
```
```suggestion
    protected ServerRequestInterface $request;
```

sfroemkenjw commented on Classes/Service/WallsService.php:
```diff
@@ -16,13 +16,13 @@
 use Psr\Http\Message\ServerRequestInterface;
 use TYPO3\CMS\Core\Core\Environment;
-use TYPO3\CMS\Core\Http\ServerRequestFactory;
+use TYPO3\CMS\Core\Http\ServerRequest;
```
```suggestion
use Psr\Http\Message\ServerRequestInterface;
```

sfroemkenjw commented on Classes/Service/WallsService.php:
```diff
@@ -55,12 +56,18 @@ class WallsService
-    public function __construct(Registry $registry, WallsIoClient $client)
+    protected ServerRequest $request;
+
+    public function __construct(Registry $registry, WallsIoClient $client, ServerRequest $request)
```
```suggestion
    public function __construct(Registry $registry, WallsIoClient $client, ServerRequestInterface $request)
```

sfroemkenjw commented on Classes/Utility/StringUtility.php:
```diff
@@ -0,0 +1,59 @@
+     * Uses native str_starts_with in PHP 8.0+, TYPO3 Core's method if available in TYPO3 v11,
+     * otherwise falls back to a custom implementation.
+     */
+    public static function beginsWith(string $haystack, string $needle): bool
```
As I had that problem a few montha ago you told me, that there are a lot of polyfills available. So, why are you implementing your own version now?

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
```diff
@@ -14,3 +14,127 @@ I'm missing new wall posts
+Emoji Handling
+==============
+
+### Description
```
It's RST and not MD ;-)

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
````diff
@@ -14,3 +14,127 @@ I'm missing new wall posts
+   ALTER TABLE index_fulltext CHANGE your_column_name your_column_name TEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
+   ```
+
+2. **Update TYPO3 Connection Configuration**
````
```suggestion
1.  **Update MySQL Database Charset and Collation**

    Run the following SQL command to update the database charset:

    ```sql
    ALTER DATABASE your_database_name CHARACTER SET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
    ```

    Update each table's charset:

    ```sql
    ALTER TABLE index_phash CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
    ALTER TABLE index_fulltext CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
    ```

    Update each text column's charset:

    ```sql
    ALTER TABLE index_phash CHANGE your_column_name your_column_name TEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
    ALTER TABLE index_fulltext CHANGE your_column_name your_column_name TEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
    ```

2.  **Update TYPO3 Connection Configuration**
```

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
````diff
@@ -14,3 +14,127 @@ I'm missing new wall posts
+
+        Edit your `LocalConfiguration.php` file:
+
+        ```php
````
This is MD, but not valif RST. Please use official syntax: https://docs.typo3.org/m/typo3/docs-how-to-document/1.0/en-us/WritingReST/Reference/Code/Codeblocks.html#quick-reference

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
```diff
@@ -14,3 +14,127 @@ I'm missing new wall posts
+
+    Update the database connection settings in TYPO3 to use `utf8mb4`:
+
+    - **For TYPO3 v9 and earlier:**
```
PLease use official "*" as list: https://docs.typo3.org/m/typo3/docs-how-to-document/1.0/en-us/WritingReST/CheatSheet.html

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
````diff
@@ -14,3 +14,127 @@ I'm missing new wall posts
+        ],
+        ```
+
+    - **For TYPO3 v10 and v11:**
````
```suggestion
    - **For TYPO3 v11:**
```

sfroemkenjw commented on Documentation/KnownProblems/Index.rst:
````diff
@@ -14,3 +14,127 @@ I'm missing new wall posts
+        ];
+        ```
+
+3. **Verify the Changes**
````
```suggestion
3.  **Verify the Changes**
```

sfroemkenjw commented on README.md:
```diff
@@ -1,18 +1,60 @@
+
+- **Server-Side Data Retrieval**: Moves all data retrieval processes to the server-side, avoiding client-side cookies.
+- **Customizable Templates**: Allows you to create your own FluidTemplate for displaying the social wall, providing full control over the design and layout.
+  (Note: Each social media provider has its own design restrctions. So, you have to adapt these design rules into your templates.)
```
```suggestion
  (Note: Each social media provider has its own design restrictions. So, you have to adapt these design rules into your templates.)
```

sfroemkenjw commented on Tests/Functional/DataProcessing/AddWallsProcessorTest.php:
```diff
@@ -96,19 +81,20 @@ public function processWithValidTypoScriptConditionWillNotModifyProcessedData():
-            ->shouldBeCalled()
+        $this->contentObjectRendererMock
+            ->method('checkIf')
+            ->with($processorConfiguration['if.'])
```
The equivalent for shouldBeCalled is missing.

```suggestion
        $this->contentObjectRendererMock
            ->expects($this->atLeastOnce())
            ->method('checkIf')
            ->with($processorConfiguration['if.'])
```

sfroemkenjw commented on Tests/Functional/DataProcessing/AddWallsProcessorTest.php:
```diff
@@ -96,19 +81,20 @@ public function processWithValidTypoScriptConditionWillNotModifyProcessedData():
+        $this->wallsServiceMock
+            ->method('getWallPosts')
+            ->with(self::isInstanceOf(PluginConfiguration::class))
+            ->willReturn([]);
```
The equivalent for shouldNotBeCalled is missing.

->expects($this->never())

sfroemkenjw commented on Tests/Functional/DataProcessing/AddWallsProcessorTest.php:
```diff
@@ -123,9 +109,9 @@ public function processWithNonPiFlexformWillAddEmptyConfVariable(): void
-        $this->wallsServiceProphecy
-            ->getWallPosts(Argument::any())
-            ->shouldBeCalled()
+        $this->wallsServiceMock
```
The equivalent for shouldBeCalled is missing.

sfroemkenjw commented on Tests/Functional/DataProcessing/AddWallsProcessorTest.php:
```diff
@@ -164,15 +150,17 @@ public function processWithValidPiFlexformWillAddConfVariable(): void
-        $this->wallsServiceProphecy
-            ->getWallPosts(Argument::any())
-            ->shouldBeCalled()
+        $this->wallsServiceMock
```
The equivalent for shouldBeCalled is missing.

sfroemkenjw commented on Tests/Functional/DataProcessing/AddWallsProcessorTest.php:
```diff
@@ -191,11 +179,13 @@ public function processWillCallGetWallPostsWithPluginConfiguration(): void
-            }))
-            ->shouldBeCalled()
+        $this->wallsServiceMock
+            ->method('getWallPosts')
```
The equivalent for shouldBeCalled is missing.

sfroemkenjw commented on Tests/Functional/DataProcessing/AddWallsProcessorTest.php:
```diff
@@ -234,9 +224,9 @@ public function processAddsWallsToProcessedData(): void
-        $this->wallsServiceProphecy
-            ->getWallPosts(Argument::any())
-            ->shouldBeCalled()
+        $this->wallsServiceMock
```
The equivalent for shouldBeCalled is missing.

sfroemkenjw commented on Tests/Functional/Fixtures/Database/be_users.csv:
```diff
@@ -0,0 +1,3 @@
+be_users,,,,,,,,,,,,,,,,,,,,
```
```suggestion
be_users,,,,
```

sfroemkenjw commented on Tests/Functional/Fixtures/Database/pages.csv:
```diff
@@ -0,0 +1,3 @@
+"pages",,,,,,,,
+,uid,pid,title,doktype,hidden,deleted,crdate,tstamp,sorting
+,1,0,Home,1,0,0,1625254800,1625254800,256
```
```suggestion
"pages",,,,,,,,,
,uid,pid,title,doktype,hidden,deleted,crdate,tstamp,sorting
,1,0,"Home",1,0,0,1625254800,1625254800,256
```

sfroemkenjw commented on Tests/Functional/Fixtures/Database/tt_content.csv:
```diff
@@ -0,0 +1,3 @@
+"tt_content",,,,,,,,,
+,uid,pid,header,bodytext,CType,hidden,deleted,crdate,tstamp,sorting
+,1,1,"Welcome to our website!","This is a sample content element.",text,0,0,1625254800,1625254800,128
```
```suggestion
"tt_content",,,,,,,,,,
,uid,pid,header,bodytext,CType,hidden,deleted,crdate,tstamp,sorting
,1,1,"Welcome to our website!","This is a sample content element.","text",0,0,1625254800,1625254800,128
```

sfroemkenjw commented on Tests/Functional/Service/WallsServiceTest.php:
```diff
@@ -15,37 +15,32 @@
+     * @var ServerRequest|MockObject|(ServerRequest&MockObject)
      */
-    protected $registry;
+    protected $requestMock;
```
```suggestion
    protected ServerRequestInterface $requestMock;
```

sfroemkenjw commented on Tests/Functional/Service/WallsServiceTest.php:
```diff
@@ -126,21 +147,24 @@ public function getWallPostsWithInvalidPluginConfigurationWillReturnEmptyArray(a
-            ->processRequest(Argument::type(PostsRequest::class))
-            ->shouldBeCalled()
+        $this->wallsIoClientMock
+            ->method('processRequest')
```
Missing expect

sfroemkenjw commented on Tests/Functional/Service/WallsServiceTest.php:
```diff
@@ -157,18 +181,20 @@ public function getWallPostsWithEmptyClientResultWillReturnCachedWalls(): void
-            ->processRequest(Argument::type(PostsRequest::class))
-            ->shouldBeCalled()
+        $this->wallsIoClientMock
+            ->method('processRequest')
```
Missing expect

sfroemkenjw commented on Tests/Functional/Service/WallsServiceTest.php:
```diff
@@ -189,25 +215,24 @@ public function getWallPostsWillReturnFreshWallPosts(): void
-            ->processRequest(Argument::type(PostsRequest::class))
-            ->shouldBeCalled()
+        $this->wallsIoClientMock
+            ->method('processRequest')
```
Missing expect

sfroemkenjw commented on Tests/Functional/Service/WallsServiceTest.php:
```diff
@@ -234,25 +259,22 @@ public function getWallPostsWillConvertTimestampHumanReadable(): void
-            ->processRequest(Argument::type(PostsRequest::class))
-            ->shouldBeCalled()
+        $this->wallsIoClientMock
+            ->method('processRequest')
```
Missing expect

sfroemkenjw commented on Tests/Functional/Service/WallsServiceTest.php:
```diff
@@ -276,25 +298,23 @@ public function getWallPostsWillConvertNewLinesToBR(): void
-            ->processRequest(Argument::type(PostsRequest::class))
-            ->shouldBeCalled()
+        $this->wallsIoClientMock
+            ->method('processRequest')
```
Missing expect

sfroemkenjw commented on Tests/Functional/ViewHelpers/Be/Security/IsAdministratorViewHelperTest.php:
```diff
@@ -11,40 +11,34 @@
 
     /**
      * @test
+     * @throws Exception
```
IMO this is deprecated. Please use $this->expectsException(\Exception::class) from within the method

sfroemkenjw commented on Tests/Functional/ViewHelpers/Be/Security/IsAdministratorViewHelperTest.php:
```diff
@@ -70,16 +64,30 @@ public function beUserIsAdministrator(): void
+        /** @var BackendUserAuthentication|MockObject $backendUserAuthenticationMock */
+        $backendUserAuthenticationMock = $this->createMock(BackendUserAuthentication::class);
+        $backendUserAuthenticationMock
+            ->method('isAdmin')
```
Missing expect

sfroemkenjw commented on Tests/Functional/ViewHelpers/Be/Security/IsAdministratorViewHelperTest.php:
```diff
@@ -70,16 +64,30 @@ public function beUserIsAdministrator(): void
+        $userAspectMock = $this->getMockBuilder(UserAspect::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $userAspectMock->method('isAdmin')->willReturn(true);
```
Missing expect

sfroemkenjw commented on Tests/Functional/ViewHelpers/Be/Security/IsAdministratorViewHelperTest.php:
```diff
@@ -70,16 +64,30 @@ public function beUserIsAdministrator(): void
+        $contextMock = $this->getMockBuilder(Context::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $contextMock->method('getAspect')->willReturn($userAspectMock);
```
Missing expect

sfroemkenjw commented on Tests/Functional/ViewHelpers/Be/Security/IsAdministratorViewHelperTest.php:
```diff
@@ -70,16 +64,30 @@ public function beUserIsAdministrator(): void
+        $contextMock->method('getAspect')->willReturn($userAspectMock);
+
+        // Replace the Context instance in the GeneralUtility
+        GeneralUtility::setSingletonInstance(Context::class, $contextMock);
```
If you're using Singletons like Context in UnitTests (not func. tests) you have to set property resetSingletonInstances to true. So, that thay all can be removed while executing parent::tearDown()

sfroemkenjw commented on Tests/Unit/Client/WallsIoClientTest.php:
```diff
@@ -11,153 +11,140 @@
         $postsRequest
-            ->isValidRequest()
-            ->shouldBeCalled()
+            ->method('isValidRequest')
```
Missing expect

sfroemkenjw commented on Tests/Unit/Client/WallsIoClientTest.php:
```diff
@@ -11,153 +11,140 @@
-            )
-            ->shouldBeCalled();
+        $this->loggerMock
+            ->expects(self::once())
```
Perfect

sfroemkenjw commented on Tests/Unit/Hook/DataHandlerTest.php:
```diff
@@ -11,44 +11,54 @@
 namespace JWeiland\WallsIoProxy\Tests\Unit\Hook;
 
+use JWeiland\WallsIoProxy\Client\WallsIoClient;
 use JWeiland\WallsIoProxy\Hook\DataHandler;
```
Is this is a huge Upgrade, can we please rename that class to DataHandlerHook?

sfroemkenjw commented on Tests/Unit/Hook/DataHandlerTest.php:
```diff
@@ -11,44 +11,54 @@
+    protected $wallsServiceMock;
+
+    /**
+     * @var ServerRequest|MockObject
```
```suggestion
     * @var ServerRequestInterface|MockObject
```

sfroemkenjw commented on composer.json:
```diff
@@ -11,14 +11,19 @@
 		"cache",
```
```suggestion
```

sfroemkenjw commented on ext_emconf.php:
```diff
@@ -1,17 +1,24 @@
     'constraints' => [
         'depends' => [
-            'typo3' => '11.5.38-12.4.99',
+            'typo3' => '11.5.39-12.4.99',
```
Thank you

sfroemkenjw commented on ext_localconf.php:
```diff
@@ -1,42 +1,69 @@
+    }
+
+    if (version_compare($versionInformation->getBranch(), '11.5', '<=')) {
+        $GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['cms/layout/class.tx_cms_layout.php']['tt_content_drawItem']['walls_io_proxy']
+            = PageLayoutViewHook::class;
```
And where is the implementation for TYPO3 12 now?

hojalatheef committed:
Update .github/workflows/ci.yml

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Update .github/workflows/ci.yml

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Update Classes/Request/AbstractRequest.php

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Update Classes/Service/WallsService.php

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Update Classes/Service/WallsService.php

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Update Tests/Unit/Hook/DataHandlerTest.php

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Update composer.json

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Update Classes/Service/WallsService.php

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
[TASK] Refactored DataHandler class to DataHandlerHook

hojalatheef committed:
Update Documentation/KnownProblems/Index.rst

Co-authored-by: Stefan Frömken <123929835+sfroemkenjw@users.noreply.github.com>

hojalatheef committed:
Removed reduntant use statement

hojalatheef committed:
[TASK] Added php 8.0 for test suite

hojalatheef committed:
[BUGFIX] Updated documentation format

hojalatheef committed:
[TASK] Revert argument type to ServerRequestInterface

hojalatheef committed:
[TASK] RequestFactory Mock issues

hojalatheef committed:
[TASK] Removed unwanted use of Class

hojalatheef committed:
[TASK] CR Fixes

hojalatheef committed:
[TASK] Fixes for CGL

hojalatheef committed:
[BUGFIX] Replaced $this->atLeastOnce() with self::atLeastOnce()

hojalatheef committed:
[BUGFIX] Replaced $this->never() with self::never()

hojalatheef committed:
[BUGFIX] Change DataHandler to DataHandlerHook in Configuration/Services.yaml

hojalatheef committed:
[TASK] Backend Preview for TYPO3 12

hojalatheef committed:
[BUGFIX] Small fixes for PHP 7.4 compatibility

hojalatheef committed:
[TASK] Namespace import optimizations

hojalatheef committed:
[TASK] Ignore stan issues related to BackendPreview
